### PR TITLE
Websocket client tests

### PR DESCRIPTION
**Pull request by adam-fowler**
Outside of the autobahn tests swift-websocket has always relied on hummingbird-websocket to test some its more obscure edge cases. This PR adds `NIOAsyncTestingChannel` based testing to extend the testing available.

### CODE REVIEW

File: Sources/WSCore/WebSocketHandler.swift [hunk 1]
```diff
@@ -272,9 +272,8 @@ public struct WebSocketCloseFrame: Sendable {
         if self.type == .client {
             frame.maskKey = self.makeMaskKey()
         }
+        self.logger.trace("Sending \(frame.traceDescription)")
```
Comment (Joannis) [inline Sources/WSCore/WebSocketHandler.swift:275]: Shouldn't the `traceDescription` be in metadata instead of interpolated? Nit though

**Review by Joannis — APPROVED** [review]

File: Sources/WSCore/WebSocketHandler.swift (hunk 1 above)
Comment (adam-fowler) [reply Sources/WSCore/WebSocketHandler.swift:275]: It hasn't changed from before. I can look at that separately
